cpp/velox/shuffle/VeloxSplitter.cc

```diff
@@ -365,115 +366,34 @@ arrow::Status VeloxSplitter::SplitFixedWidthValueBuffer(const velox::RowVector&
   for (auto col = 0; col < fixed_width_column_count_; ++col) {
     auto col_idx = simple_column_indices_[col];
     auto column = rv.childAt(col_idx);
-    assert(column->isFlatEncoding());
 
-    const uint8_t* src_addr = (const uint8_t*)column->valuesAsVoid();
+    velox::DecodedVector dv(*column);
```

Comment:
It is suggested to save `velox::DecodedVector dv`, and then invoke dv.decode(*column) for each column

Reply:
I used another way

Reply:
In this way, we will initialize class `velox::DecodedVector` with each Batch, but for velox purpose, We should create one `velox::DecodedVector`  for all Batches, https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/aggregates/ApproxDistinctAggregate.cpp#L377

Reply:
OK